packages/engine/src/lib.rs

```diff
@@ -1,21 +1,32 @@
-#![feature(test)]
-#![feature(map_try_insert)]
-#![feature(box_patterns)]
-#![feature(once_cell)]
-#![feature(backtrace)]
-
-#[macro_use]
-extern crate lazy_static;
-
-#[macro_use]
-extern crate derive_new;
+#![feature(backtrace, box_patterns, map_try_insert, once_cell, test)]
+#![warn(
+    rust_2018_compatibility,
+    rust_2018_idioms,
+    rust_2021_compatibility,
+    deprecated_in_future,
+    elided_lifetimes_in_paths,
+    explicit_outlives_requirements,
+    keyword_idents,
+    macro_use_extern_crate,
+    meta_variable_misuse,
+    missing_abi,
+    non_ascii_idents,
+    noop_method_call,
+    pointer_structural_match,
+    // Do we want this?
```

Comment:
Appreciate you also wrote it in the PR, but just so it doesn't get lost in case we don't handle it before merging
```suggestion
    // TODO: Do we want this?
```

Reply:
Actually I wanted to remove the two lines before merging. 😄